core/src/main/kotlin/org/evomaster/core/output/dto/DtoWriter.kt

```diff
@@ -69,34 +69,49 @@ class DtoWriter {
     }
 
     private fun calculateDtoFromChoice(gene: ChoiceGene<*>, actionName: String) {
-        val dtoName = TestWriterUtils.safeVariableName(actionName)
-        val dtoClass = DtoClass(dtoName)
-        val children = gene.getViewOfChildren()
-        // merge options into a single DTO
-        children.forEach { childGene ->
-            when (childGene) {
-                is ObjectGene -> populateDtoClass(dtoClass, childGene)
-                is ArrayGene<*> -> {
-                    val template = childGene.template
-                    if (template is ObjectGene) {
-                        populateDtoClass(dtoClass, template)
+        // TODO: should we handle EnumGene?
+        if (hasObjectOrArrayGene(gene)) {
```

Comment:
@arcuri82, added this check instead of checking for the first gene inside a `ChoiceGene` since:
1. When testing, `EnumGene` came as the first one (and also the active one)
2. Since we'll be merging `ObjectGene`s and `ArrayGene`s (lines below) it seemed the most reasonable check

What do you think?